framework/captcha/Driver.php

```diff
@@ -0,0 +1,79 @@
+<?php
+/**
+ * @link http://www.yiiframework.com/
+ * @copyright Copyright (c) 2008 Yii Software LLC
+ * @license http://www.yiiframework.com/license/
+ */
+
+namespace yii\captcha;
+
+use Yii;
+use yii\base\Component;
+use yii\base\InvalidConfigException;
+
+/**
+ * Driver is the base class for CAPTCHA rendering driver classes.
+ *
+ * By configuring the properties of Driver, you may customize the appearance of
+ * the generated CAPTCHA images, such as the font color, the background color, etc.
+ *
+ * @author Paul Klimov <klimov.paul@gmail.com>
+ * @author Qiang Xue <qiang.xue@gmail.com>
+ * @since 2.1.0
+ */
+abstract class Driver extends Component implements DriverInterface
+{
+    use VerifyCodeGeneratorTrait;
```

Comment:
Trait might ve needed for particular developer-defined driver, which renders image with other options or format, but wishes to use existing code generation.